include/swift/AST/LocalizationFormat.h

```diff
@@ -0,0 +1,111 @@
+//===--- LocalizationFormat.h - YAML format for Diagnostic Messages ---*-
+// C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the format for localized diagnostic messages.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_LOCALIZATIONFORMAT_H
+#define SWIFT_LOCALIZATIONFORMAT_H
+
+#include "swift/AST/DiagnosticsCommon.h"
+#include "llvm/Support/YAMLParser.h"
+#include "llvm/Support/YAMLTraits.h"
+
+enum class swift::DiagID : uint32_t {
+#define DIAG(KIND, ID, Options, Text, Signature) ID,
+#include "swift/AST/DiagnosticsAll.def"
+};
+
+struct DiagnosticNode {
+  swift::DiagID id;
+  std::string msg;
+};
+
+namespace llvm {
+namespace yaml {
+
+template <> struct ScalarEnumerationTraits<swift::DiagID> {
+  static void enumeration(IO &io, swift::DiagID &value) {
+#define DIAG(KIND, ID, Options, Text, Signature)                               \
+  io.enumCase(value, #ID, swift::DiagID::ID);
+#include "swift/AST/DiagnosticsAll.def"
+  }
+};
+
+template <> struct MappingTraits<DiagnosticNode> {
+  static void mapping(IO &io, DiagnosticNode &node) {
+    io.mapRequired("id", node.id);
+    io.mapRequired("msg", node.msg);
+  }
+};
+
+} // namespace yaml
+} // namespace llvm
+
+LLVM_YAML_IS_SEQUENCE_VECTOR(DiagnosticNode)
+
+class LocalizationInput : public llvm::yaml::Input {
```

Comment:
I wasn't able to separate the declaration and the implementation of this class's methods in two files i.e. `.h` and `.cpp` files as it was causing build failed related to `llvm::yaml`. So, I decided to put everything in the `.h` files for now. WDYT?

Reply:
@HassanElDesouky what was exactly the issue? Some unqualified names or missing using namespace maybe?
For now, this may be fine because the methods are not large, but it this is expected to grow it may be good to put in the `.cpp` :)

Reply:
No this won’t grow. I was just overriding a method and an operator from `llvm::yaml::Input`